samples/extensions/dynamic_rendering/dynamic_rendering.cpp

```diff
@@ -0,0 +1,425 @@
+	    vkb::initializers::pipeline_dynamic_state_create_info(
+	        dynamic_state_enables.data(),
+	        static_cast<uint32_t>(dynamic_state_enables.size()),
+	        0);
+
+	// Vertex bindings an attributes for model rendering
+	// Binding description
+	std::vector<VkVertexInputBindingDescription> vertex_input_bindings = {
+	    vkb::initializers::vertex_input_binding_description(0, sizeof(Vertex), VK_VERTEX_INPUT_RATE_VERTEX),
+	};
+
+	// Attribute descriptions
+	std::vector<VkVertexInputAttributeDescription> vertex_input_attributes = {
+	    vkb::initializers::vertex_input_attribute_description(0, 0, VK_FORMAT_R32G32B32_SFLOAT, 0),                        // Position
+	    vkb::initializers::vertex_input_attribute_description(0, 1, VK_FORMAT_R32G32B32_SFLOAT, sizeof(float) * 3),        // Normal
+	};
+
+	VkPipelineVertexInputStateCreateInfo vertex_input_state = vkb::initializers::pipeline_vertex_input_state_create_info();
+	vertex_input_state.vertexBindingDescriptionCount        = static_cast<uint32_t>(vertex_input_bindings.size());
+	vertex_input_state.pVertexBindingDescriptions           = vertex_input_bindings.data();
+	vertex_input_state.vertexAttributeDescriptionCount      = static_cast<uint32_t>(vertex_input_attributes.size());
+	vertex_input_state.pVertexAttributeDescriptions         = vertex_input_attributes.data();
+
+	std::array<VkPipelineShaderStageCreateInfo, 2> shader_stages{};
+	shader_stages[0] = load_shader("dynamic_rendering/gbuffer.vert", VK_SHADER_STAGE_VERTEX_BIT);
+	shader_stages[1] = load_shader("dynamic_rendering/gbuffer.frag", VK_SHADER_STAGE_FRAGMENT_BIT);
+
+	// Create graphics pipeline for dynamic rendering
+	VkFormat color_rendering_format = VK_FORMAT_R8G8B8A8_UNORM;
+
+	// Provide information for dynamic rendering
+	VkPipelineRenderingCreateInfoKHR pipeline_create{VK_STRUCTURE_TYPE_PIPELINE_RENDERING_CREATE_INFO_KHR};
+	pipeline_create.pNext                   = VK_NULL_HANDLE;
+	pipeline_create.colorAttachmentCount    = 1;
+	pipeline_create.pColorAttachmentFormats = &color_rendering_format;
+	pipeline_create.depthAttachmentFormat   = depth_format;
+	pipeline_create.stencilAttachmentFormat = depth_format;
+
+	// Use the pNext to point to the rendering create struct
+	VkGraphicsPipelineCreateInfo graphics_create{VK_STRUCTURE_TYPE_GRAPHICS_PIPELINE_CREATE_INFO};
+	graphics_create.pNext               = &pipeline_create;
+	graphics_create.renderPass          = VK_NULL_HANDLE;
+	graphics_create.pInputAssemblyState = &input_assembly_state;
+	graphics_create.pRasterizationState = &rasterization_state;
+	graphics_create.pColorBlendState    = &color_blend_state;
+	graphics_create.pMultisampleState   = &multisample_state;
+	graphics_create.pViewportState      = &viewport_state;
+	graphics_create.pDepthStencilState  = &depth_stencil_state;
+	graphics_create.pDynamicState       = &dynamic_state;
+	graphics_create.pVertexInputState   = &vertex_input_state;
+	graphics_create.stageCount          = shader_stages.size();
```

Comment:
Very minor: I think this is missing an explicit cast, which gives warning C4267 in MSVC.